components/builder-api/src/server/resources/events.rs

```diff
@@ -0,0 +1,179 @@
+use crate::{db::models::channel::{AuditPackage,
+                                  AuditPackageEvent,
+                                  ListEvents},
+            server::{authorize::authorize_session,
+                     error::{Error,
+                             Result},
+                     framework::headers,
+                     helpers::{self,
+                               req_state,
+                               DateRange,
+                               Pagination,
+                               ToChannel},
+                     AppState}};
+use actix_web::{http::{self,
+                       HeaderMap},
+                web::{self,
+                      Data,
+                      Query,
+                      ServiceConfig},
+                HttpRequest,
+                HttpResponse};
+use builder_core::http_client::{HttpClient,
+                                USER_AGENT_BLDR};
+
+pub struct Events {}
+
+impl Events {
+    // Route registration
+    //
+    pub fn register(cfg: &mut ServiceConfig) {
+        cfg.route("/depot/events", web::get().to(get_events))
+           .route("/depot/events/saas", web::get().to(get_events_from_saas));
+    }
+}
+
+#[allow(clippy::needless_pass_by_value)]
+async fn get_events(req: HttpRequest,
+                    pagination: Query<Pagination>,
+                    channel: Query<ToChannel>,
+                    date_range: Query<DateRange>)
+                    -> HttpResponse {
+    match do_get_events(&req, &pagination, &channel, &date_range) {
+        Ok((events, count)) => postprocess_event_list(&req, &events, count, &pagination),
+        Err(err) => {
+            error!("{}", err);
+            err.into()
+        }
+    }
+}
+
+#[allow(clippy::needless_pass_by_value)]
+async fn get_events_from_saas(req: HttpRequest,
+                              pagination: Query<Pagination>,
+                              channel: Query<ToChannel>,
+                              date_range: Query<DateRange>,
+                              state: Data<AppState>)
+                              -> HttpResponse {
+    let bldr_url = &state.config.api.saas_bldr_url;
+    let headers = req.headers();
+    return get_events_from_saas_builder(headers,
```

Comment:
I think you can consolidate `get_events_from_saas_builder` into this function now.